src/Collapse.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import omit from 'lodash.omit';
+
+const SHOW = 'SHOW';
+const SHOWN = 'SHOWN';
+const HIDE = 'HIDE';
+const HIDDEN = 'HIDDEN';
+
+const propTypes = {
+  isOpen: PropTypes.bool,
+  className: PropTypes.node
+};
+function reflow(element) {
+  ((function (bs) { return bs; })(element.offsetHeight));
+}
+const defaultProps = { isOpen: false };
+
+class Collapse extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      collapse: props.isOpen ? SHOWN : HIDDEN,
+    };
+    this.element = null;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const willOpen = nextProps.isOpen;
+    const collapse = this.state.collapse;
+
+    if (willOpen && collapse === HIDDEN) {
+      // will open
+      this.setState({ collapse: SHOW }, () => {
+        // start transition
+        this.element.style.height = `${this.getHeight()}px`;
```

Comment:
I thought the timing will be off at first, now I am trying to move these dom attribute changes to react. Let's see if it will work fine.